### PR TITLE
fix(hz): resolve typedef panic for thrift

### PR DESCRIPTION
**Pull request by FGYFFFF**
#### What type of PR is this?

fix
#### Check the PR title.

- [x] This PR title match the format: `<type>(optional scope): <description>`.
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io).

#### (Optional) Translate the PR title into Chinese.
解析请求/返回类型遇到 typedef 出现预期之外的 panic

#### (Optional) More detail description for this PR(en: English/zh: Chinese).
 
en: resolve request/response type encounters a typedef with an unexpected panic
zh(optional): 解析请求/返回类型遇到 typedef 出现预期之外的 panic

### CODE REVIEW

File: cmd/hz/thrift/resolver.go [hunk 1]
```diff
@@ -285,6 +285,14 @@ func (resolver *Resolver) ResolveIdentifier(id string) (ret ResolvedSymbol, err
 func (resolver *Resolver) ResolveTypeName(typ *parser.Type) (string, error) {
 	switch typ.GetCategory() {
 	case parser.Category_Map:
+		if typ.GetIsTypedef() {
```
Comment (welkeyever) [inline cmd/hz/thrift/resolver.go:288]: 这块代码移到switch case之上，只需要执行一次？

Comment (FGYFFFF) [reply cmd/hz/thrift/resolver.go]: done~

**Comment by welkeyever** [comment]
Checkbox in description.

File: cmd/hz/thrift/resolver.go [hunk 2]
```diff
@@ -283,6 +283,14 @@ func (resolver *Resolver) ResolveIdentifier(id string) (ret ResolvedSymbol, err
 }
 
 func (resolver *Resolver) ResolveTypeName(typ *parser.Type) (string, error) {
+	if typ.GetIsTypedef() {
+		rt, err := resolver.ResolveIdentifier(typ.GetName())
+		if err != nil {
+			return "", err
+		}
+
+		return rt.Expression(), nil
```
Comment (Duslia) [inline cmd/hz/thrift/resolver.go:292]: return rt.Expression(), err ?

**Review by li-jin-gou — APPROVED** [review]

**Review by welkeyever — APPROVED** [review]

**Review by Duslia — APPROVED** [review]

**Review by wzekin — APPROVED** [review]

**Pull request merged**